web_interface/blueprints/api_v3.py

```
@@ -2846,6 +2853,79 @@ def update_plugin():
+
+    # Record whether the plugin was running before we touch anything.
+    was_loaded = (
+        api_v3.plugin_manager is not None
+        and plugin_id in api_v3.plugin_manager.plugins
+    )
+
+    def _rollback(reload_plugin):
+        if main_snapshot is not None:
+            try:
+                api_v3.config_manager.save_raw_file_content('main', main_snapshot)
+            except Exception as restore_err:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)